torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -111,29 +122,73 @@ def create_subclass_meta(
 # a list of tensors that we would then need to concat together.
 # Instead, we specialize the logic for the inference vs. joint graph case.
 # NOTE: this function is hot, since we unwrap tensor subclass inputs at runtime
-def unwrap_tensor_subclasses(wrapped_args, *, is_joint_structure: bool):
-    def concat_inner_tensors_from_subclasses(xs):
-        xs_inner = []
-        for x in xs:
-            if is_traceable_wrapper_subclass(x):
-                xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
-            else:
-                xs_inner.append(x)
-        return xs_inner
 
-    if is_joint_structure:
-        assert isinstance(wrapped_args, tuple) and len(wrapped_args) == 2
-        assert isinstance(wrapped_args[0], (tuple, list)) and isinstance(
-            wrapped_args[1], (tuple, list)
-        )
-        unwrapped_args_fw = concat_inner_tensors_from_subclasses(wrapped_args[0])
-        unwrapped_args_tangents = concat_inner_tensors_from_subclasses(wrapped_args[1])
-        unwrapped_args = (unwrapped_args_fw, unwrapped_args_tangents)
-    else:
-        assert isinstance(wrapped_args, (list, tuple))
-        unwrapped_args_fw = concat_inner_tensors_from_subclasses(wrapped_args)
-        unwrapped_args = unwrapped_args_fw
-    return unwrapped_args
+
+def filter_symints(lst: Iterable[Union[int, SymInt]]) -> List[SymInt]:
+    return [s for s in lst if isinstance(s, SymInt) and not s.node.is_nested_int()]
+
+
+def compute_symint_placeholders(lst: Iterable[Union[int, SymInt]]) -> List[bool]:
+    return [isinstance(s, SymInt) and not s.node.is_nested_int() for s in lst]
```

Comment:
hmm - this function can be called at runtime.

The problem is that:
(1) for @jamesjwu 's warm caching work, we need to guarantee that all metadata that we need at runtime is serializable
(2) `SymInts` are not serializable - so in order for your PR to work with James's cache, you're going to need to update `SubclassCreationMeta.make_runtime_safe()` to scrub out the SymInts. One option is to replace every SymInt in the `outer_size/outer_stride` with `None`
(3) you'll then need to update this code, since it won't be able to check for SymInt types at runtime

One wart is the NJT nested_int stuff: per @soulitzer 's comment, we *do not* want AOTDispatcher to grab the outer_sizes/stride from the graph outputs when we have an NJT output (and NJT actually [ignores](https://github.com/pytorch/pytorch/blob/main/torch/nested/_internal/nested_tensor.py#L275) the`outer_size/stride` arguments when the inputs to unflatten are not fake tensors).

One easy option would be: in `SubclassCreationMeta.make_runtime_safe()` when you are scrubbing out SymInts and replacing them with `None`, if you see a symint that is a nested int (`s.node.is_nested_int()`) - instead of replacing it with `None`, replace it with a dummy int like `-1` (that is not a valid size). This should properly instruct AOTDispatcher at runtime to **not** try to grab the outer_size as a graph output SymInt, and just plumb the (plain int) `-1` as the outer size at runtime. Which should be fine, since NJT will ignore it and use the `offsets` that gets passed in to construct the size.